qualtran/symbolics/math_funcs_test.py

```diff
@@ -16,15 +16,31 @@
 import numpy as np
 import pytest
 import sympy
+from sympy.codegen.cfunctions import log
 from sympy.codegen.cfunctions import log2 as sympy_log2
 
-from qualtran.symbolics import bit_length, ceil, is_zero, log2, sarg, sexp, smax, smin
+from qualtran.symbolics import bit_length, ceil, is_zero, ln, log2, sarg, sexp, smax, smin, ssqrt
 
 
 def test_log2():
     assert log2(sympy.Symbol('x')) == sympy_log2(sympy.Symbol('x'))
     assert log2(sympy.Number(10)) == sympy_log2(sympy.Number(10))
     assert log2(10) == np.log2(10)
+    assert log2(2**10000) == 10000
+
+
+def test_ln():
+    assert ln(sympy.Symbol('x')) == log(sympy.Symbol('x'))
+    assert ln(sympy.Number(10)) == log(sympy.Number(10))
+    assert ln(10) == np.log(10)
+    assert np.isclose(ln(2**1000) / ln(2), 1000)
+
+
+def test_ssqrt():
+    assert ssqrt(sympy.Symbol('x')) == sympy.sqrt(sympy.Symbol('x'))
+    assert ssqrt(sympy.Number(10)) == sympy.sqrt(sympy.Number(10))
+    assert ssqrt(10) == np.sqrt(10)
+    assert ssqrt(2**1000) == 2**500
```

Comment:
`math.sqrt` fails for larger inputs e.g. 
```py3
>>> math.sqrt(2**10000)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
OverflowError: int too large to convert to float
```

Reply:
This is because return type of `math.sqrt()` is a float, but its input types can be large integers. `np.sqrt()` fails even on 2 ** 100 because it expects the input type to be 64 bit integers at-most.